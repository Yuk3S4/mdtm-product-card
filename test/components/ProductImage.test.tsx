import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('Pruebas en el <ProductImage />', () => { 
    test('debe de mostrar el componente correctamente con la imagen', () => { 
        
        const wrapper = renderer.create(
            <ProductImage img="https://img1.png" />    
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
        

    });

    test('debe de mostrar el componente con la imagen del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage />    
                    )
                }
            </ProductCard>    
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    })
})