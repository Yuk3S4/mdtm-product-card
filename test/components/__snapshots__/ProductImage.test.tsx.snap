// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <ProductImage /> debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en el <ProductImage /> debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<img
  alt="Product Image"
  className="productImg undefined"
  src="https://img1.png"
/>
`;

exports[`Pruebas en el <ProductImage /> debe de mostrar el componente correctamente con la imagen 1`] = `
<img
  alt="Product Image"
  className="productImg undefined"
  src="https://img1.png"
/>
`;
